lib/course/middleware/etag.rb

```diff
@@ -0,0 +1,38 @@
+module Course
+  module Middleware
+    class Etag
+      attr_reader :app
+
+      def initialize(app)
+        @app = app
+      end
+
+      def call(env)
```

Comment:
no, you got it wrong
please read carefully how etag works.
Yes we need to calculated digest, this is correct.

PS. This middleware expect `digest` lib to be required, we should put it right here, not elsewhere.